### DISCUSSION TITLE
How to call middleware only on the client?

### DISCUSSION THREAD

**Discussion by nikableh** [nuxt/nuxt#25665] [Questions] [answered]
I have a middleware `auth` that redirects you from page `/account` either to `/auth` or to `/user/[id]` depending on if you have a WT token inside the `localStorage`.

Obviously, this middleware can run only on the client because it needs access to `localStorage`. And I've successfully achieved that with this check at the beginning of the middleware:
```ts
if (process.server) {
  return;
}
```

But then there is a weird problem that visiting `/account` route directly causes a hydration mismatch warning. Funnily enough, this warning doesn't appear when you visit `/account` through `<NuxtLink/>`.

What cam be the problem?

**Comment by nikableh** [comment]
This is the middleware in question:
```ts
import { jwtDecode } from "jwt-decode";

export default defineNuxtRouteMiddleware((to, from) => {
  if (process.server) {
    return;
  }

  const token = localStorage.getItem("jwtToken");

  if (!token) {
    return navigateTo("/auth");
  }

  const tokenPayload = jwtDecode(token);
  return navigateTo(`/user/${tokenPayload.sub}`);
});
```

**Comment by nikableh** [comment, marked as answer]
Solved by disabling SSR for `/account` route. Are there any better solutions?
``` js
routeRules: {
  '/account': { ssr: false },
}
```

I will mark this as an answer a bit later in case there are better ways to achieve what I'm trying to do.

**Comment by alvinindra** [comment]
I think it's recommended to `useCookie` for the token so you can that in SSR

https://nuxt.com/docs/api/composables/use-cookie
